projects/go-style-guide/src/app/features/ui-kit/components/table-docs/components/table-title-template/table-title-template.component.html

```diff
@@ -0,0 +1,41 @@
+<section class="go-container">
+
+  <go-card class="go-column go-column--100">
+    <ng-container go-card-header>
+      <h2 class="go-heading-2">Implementing a Title Template</h2>
+    </ng-container>
+    <ng-container go-card-content>
+      <p class="go-body-copy">
+        It is possible to add a custom template for the table title
+        by using <code class="code-block--inline">#goTableTitle</code>.
+      </p>
+      <p class="go-body-copy">
+        The content within these areas can be any HTML and is supported by
+        Angular so you can make use of components, pipes, etc.
+      </p>
+      <p class="go-body-copy">
+        <strong>
+          If a table title template is used, it will replace the normal table title area.
+        </strong>
+      </p>
+      <code [highlight]="headerTemplating" class="code-block--no-bottom-margin"></code>
+    </ng-container>
+  </go-card>
+
+  <go-table
+    class="go-column go-column--100"
+    [tableConfig]="tableConfig">
+    <ng-template #goTableTitle>
+      <go-button (handleClick)="makeToast()">
+        Make Toast
+      </go-button>
```

Comment:
Should we maybe make this a little more realistic of an example? This seems like a good test case to make sure it works, but I don't see this actually being used like this. Maybe make the example be a header with a go-icon-button beside it that when clicked makes a toast or something?

Reply:
That's fair, I've updated.